From c0309c5963a030ccecb38fd189920665af2cc35b Mon Sep 17 00:00:00 2001
From: itsmohitj <itsmohitj@gmail.com>
Date: Fri, 24 Jul 2020 12:24:42 +0530
Subject: [PATCH] Created Patch File

---
 client3.py | 42 +++++++++++++++++++++++-------------------
 1 file changed, 23 insertions(+), 19 deletions(-)

diff --git a/client3.py b/client3.py
index f1771c3..3f74dd9 100644
--- a/client3.py
+++ b/client3.py
@@ -30,30 +30,34 @@ QUERY = "http://localhost:8080/query?id={}"
 N = 500
 
 def getDataPoint(quote):
-	""" Produce all of the needed values to generate a datapoint """
-	""" ------------- Update this function ------------- """
-	stock = quote['stock']
-	bid_price = float(quote['top_bid']['price'])
-	ask_price = float(quote['top_ask']['price'])
-	price = bid_price
-	return stock, bid_price, ask_price, price
+        """ Produce all of the needed values to generate a datapoint """
+        """ ------------- Update this function ------------- """
+        stock = quote['stock']
+        bid_price = float(quote['top_bid']['price'])
+        ask_price = float(quote['top_ask']['price'])
+        price = (bid_price + ask_price) + 2
+        return stock, bid_price, ask_price, price
 
 def getRatio(price_a, price_b):
-	""" Get ratio of price_a and price_b """
-	""" ------------- Update this function ------------- """
-	""" Also create some unit tests for this function in client_test.py """
-	return 1
+        """ Get ratio of price_a and price_b """
+        """ ------------- Update this function ------------- """
+        """ Also create some unit tests for this function in client_test.py """
+        if(price_b==0):
+                return
+        return price_a/price_b
 
 # Main
 if __name__ == "__main__":
 
-	# Query the price once every N seconds.
-	for _ in iter(range(N)):
-		quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
+        # Query the price once every N seconds.
+        for _ in iter(range(N)):
+                quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
 
-		""" ----------- Update to get the ratio --------------- """
-		for quote in quotes:
-			stock, bid_price, ask_price, price = getDataPoint(quote)
-			print ("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
+                """ ----------- Update to get the ratio --------------- """
+                prices = {}
+                for quote in quotes:
+                        stock, bid_price, ask_price, price = getDataPoint(quote)
+                        prices[stock] = price
+                        print ("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-		print ("Ratio %s" % getRatio(price, price))
+                print ("Ratio %s" % getRatio(prices['ABC'], prices['DEF']))
-- 
2.25.1

